pkg/scalers/activemq_scaler_test.go

```diff
@@ -0,0 +1,112 @@
+	// Missing management endpoint should fail
+	{map[string]string{"destinationName": "testQueue", "brokerName": "localhost", "targetQueueSize": "10", "metricName": "testMetricName"}, true, map[string]string{"username": "testUsername", "password": "pass123"}},
+	// Missing destination name, should fail
+	{map[string]string{"managementEndpoint": "localhost:8161", "brokerName": "localhost", "targetQueueSize": "10", "metricName": "testMetricName"}, true, map[string]string{"username": "testUsername", "password": "pass123"}},
+	// Missing broker name, should fail
+	{map[string]string{"managementEndpoint": "localhost:8161", "destinationName": "testQueue", "targetQueueSize": "10", "metricName": "testMetricName"}, true, map[string]string{"username": "testUsername", "password": "pass123"}},
+	// Missing username, should fail
+	{map[string]string{"managementEndpoint": "localhost:8161", "destinationName": "testQueue", "brokerName": "localhost", "targetQueueSize": "10", "metricName": "testMetricName"}, true, map[string]string{"password": "pass123"}},
+	// Missing password, should fail
+	{map[string]string{"managementEndpoint": "localhost:8161", "destinationName": "testQueue", "brokerName": "localhost", "targetQueueSize": "10", "metricName": "testMetricName"}, true, map[string]string{"username": "testUsername"}},
+	// Properly formed metadata with restAPITemplate
+	{map[string]string{"restAPITemplate": "http://localhost:8161/api/jolokia/read/org.apache.activemq:type=Broker,brokerName=localhost,destinationType=Queue,destinationName=testQueue/QueueSize", "targetQueueSize": "10", "metricName": "testMetricName"}, false, map[string]string{"username": "testUsername", "password": "pass123"}},
+	// Invalid restAPITemplate, should fail
+	{map[string]string{"restAPITemplate": testInvalidRestAPITemplate, "targetQueueSize": "10", "metricName": "testMetricName"}, true, map[string]string{"username": "testUsername", "password": "pass123"}},
+	// Missing username, should fail
+	{map[string]string{"restAPITemplate": "http://localhost:8161/api/jolokia/read/org.apache.activemq:type=Broker,brokerName=localhost,destinationType=Queue,destinationName=testQueue/QueueSize", "targetQueueSize": "10", "metricName": "testMetricName"}, true, map[string]string{"password": "pass123"}},
+	// Missing password, should fail
+	{map[string]string{"restAPITemplate": "http://localhost:8161/api/jolokia/read/org.apache.activemq:type=Broker,brokerName=localhost,destinationType=Queue,destinationName=testQueue/QueueSize", "targetQueueSize": "10", "metricName": "testMetricName"}, true, map[string]string{"username": "testUsername"}},
+}
+
+func TestActiveMQParseMetadata(t *testing.T) {
+	for _, testData := range testActiveMQMetadata {
+		metadata, err := parseActiveMQMetadata(&ScalerConfig{TriggerMetadata: testData.metadata, AuthParams: testData.authParams})
```

Comment:
it might be helpful to put the code inside this for loop inside a `t.Run("test name", func(t *testing.T) { test_code_here })` block. that way, test failures will be labeled with the individual test case that failed, which can help debugging (especially debugging CI failures going forward). 

the same can be said of the other table-driven tests below

Reply:
Sorry, I forgot to mention: I don't think this is absolutely necessary to do, just a suggestion if you think it would help